src/main/cpp/src/get_json_object.cu

```diff
@@ -988,49 +1016,22 @@ bool check_error(cudf::detail::host_vector<int8_t> const& error_check)
     error_check.cbegin(), error_check.cend(), [](auto const val) { return val != 0; });
 }
 
-std::vector<std::unique_ptr<cudf::column>> get_json_object(
+std::vector<std::unique_ptr<cudf::column>> get_json_object_batch(
   cudf::strings_column_view const& input,
-  std::vector<std::vector<std::tuple<path_instruction_type, std::string, int32_t>>> const&
+  cudf::detail::input_offsetalator const& in_offsets,
+  std::vector<cudf::host_span<std::tuple<path_instruction_type, std::string, int32_t> const>> const&
     json_paths,
+  int64_t scratch_size,
   rmm::cuda_stream_view stream,
   rmm::device_async_resource_ref mr)
 {
-  auto const num_outputs = json_paths.size();
-  std::vector<std::unique_ptr<cudf::column>> output;
-
-  // Input is empty or all nulls - just return all null columns.
-  if (input.is_empty() || input.size() == input.null_count()) {
-    for (std::size_t idx = 0; idx < num_outputs; ++idx) {
-      output.emplace_back(std::make_unique<cudf::column>(input.parent(), stream, mr));
-    }
-    return output;
-  }
-
-  auto const d_input_ptr = cudf::column_device_view::create(input.parent(), stream);
-  auto const in_offsets =
-    cudf::detail::offsetalator_factory::make_input_iterator(input.offsets(), input.offset());
   auto const [d_json_paths, h_json_paths, d_inst_names, h_inst_names] =
     construct_path_commands(json_paths, stream);
 
-  auto const max_row_size = thrust::transform_reduce(
-    rmm::exec_policy(stream),
-    thrust::make_counting_iterator(0),
-    thrust::make_counting_iterator(input.size()),
-    cuda::proclaim_return_type<int64_t>(
-      [in_offsets] __device__(auto const idx) { return in_offsets[idx + 1] - in_offsets[idx]; }),
-    int64_t{0},
-    thrust::maximum{});
+  auto const num_outputs = json_paths.size();
+  std::vector<std::unique_ptr<cudf::column>> output;
 
-  // We will use scratch buffers to store the output strings without knowing their sizes.
-  // Since we do not know their sizes, we need to allocate the buffer a bit larger than the input
-  // size so that we will not write output strings into an out-of-bound position.
-  // Checking out-of-bound needs to be performed in the main kernel to make sure we will not have
-  // data corruption.
-  auto const scratch_size = [&, max_row_size = max_row_size] {
-    // Pad the scratch buffer by an additional size that is a multiple of max row size.
-    auto constexpr padding_rows = 10;
-    return input.chars_size(stream) + max_row_size * padding_rows;
-  }();
+  auto const d_input_ptr = cudf::column_device_view::create(input.parent(), stream);
```

Comment:
Sorry, I missed this. `d_input_ptr` can be at the caller (in `get_json_object`). Then it is passed in this by:
```
std::vector<std::unique_ptr<cudf::column>> get_json_object_batch(
  cudf::column_device_view const& input,
)
```
At the caller:
```
auto tmp = get_json_object_batch(*d_input_ptr, in_offsets, batch, scratch_size, stream, mr);
```

This is because each time calling to `column_device_view::create` is also not trivial.